### PR TITLE
Default port to 8080 when missing

### PR DESCRIPTION
**Pull request by andrewballantyne**
https://jira.coreos.com/browse/ODC-1727

- Reworked PR for 4.2 - simply to unblock the route creation when no port is exposed
- Only UI change is the addition of a text field when expanding Routing Advanced Options
- The default port is 8080 if we do not find one when searching the image ref

### CODE REVIEW

**Comment by spadgett** [comment]
I'm a little confused by this change. It seems like in most cases this default won't work. Are we always creating the service?

**Comment by andrewballantyne** [comment]
> I'm a little confused by this change. It seems like in most cases this default won't work. Are we always creating the service?

@christianvogt @sspeiche and I talked about how to make the shortest path for the dev to creating the route. If 8080 is not what they want, they can add the Routing advanced section and change it to a port of their choosing.

8080 was suggested as the default as a means to get this unblocked for 4.2 - as without this change there is no way to create a route if it's not exposed in your image.

For 4.3, we plan to discuss with UX to find out a better flow.

File: frontend/packages/dev-console/src/components/import/__mocks__/deployImage-validation-mock.ts [hunk 1]
```diff
@@ -56,14 +56,14 @@ export const mockFormData: DeployImageFormData = {
     },
     name: 'myimage',
     tag: 'latest',
-    status: 'success',
+    status: { metadata: {}, status: 'success' },
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/import/__mocks__/deployImage-validation-mock.ts]: The bad type for status was used by a previous PR, so I had to update it once I rebased - didn't catch the typescript error when I rebased on master.

**Comment by christianvogt** [comment]
> I'm a little confused by this change. It seems like in most cases this default won't work. Are we always creating the service?

@spadgett No we are not always creating the service. Only when the user chooses to create a route. If the user wants to create a route, but there is no exposed port we give them the option to now enter the port by text input instead of what is normally a dropdown containing the exposed ports. The default is 8080 but the user can change it.

Several options were discussed and providing a default for now was the lesser change without requiring design to re-flow the form fields.

**Comment by andrewballantyne** [comment]
/test e2e-aws-console-olm

They appear unique and not what I was getting on my last PR. 🤞 that this eventually passes.

**Pull request merged**